proposals/streams/microprofile-streams-api/src/main/java/org/eclipse/microprofile/streams/Ingest.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.microprofile.streams;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Methods annotated with @Ingest must return a {@link java.util.concurrent.Flow.Subscriber} to consume messages on
```

Comment:
Can this be used for publish and consume? Maybe add another method to differentiate it.

Reply:
Yes, we would add an `@Egress` for publishing. I'm very interested in what people think of that name choice, as I mentioned on the mailing list, it's only a recent idea to use these names.

Reply:
Btw, I would imagine you'd be able to do something like this:

```java
@Ingress(topic = "topic1")
@Egress(topic = "topic2")
Processor<Envelope<Topic1Message>, Envelope<Topic2Message>> processStream() {
  return Flow.map(envelope ->
      envelope.withPayload(transform(envelope.getPayload()));
  ).asProcessor().run(materializer);
}

private Topic2Message transform(Topic2Message msg) {
  ...
}
```